docker-app/worker_wrapper/wrapper.py

```diff
@@ -522,3 +547,31 @@ def after_docker_exception(self) -> None:
         if project.project_details is not None:
             project.project_details = None
             project.save(update_fields=("project_details",))
+
+
+def cancel_orphaned_workers():
+    client: DockerClient = docker.from_env()
+
+    running_workers: List[Container] = client.containers.list(
+        filters={
+            "label": "worker"
+        }
+    )
+
+    worker_ids = [c.id for c in running_workers]
+
+    worker_with_job_ids = Job.objects.filter(container_id__in=worker_ids).values_list(
+        "container_id", flat=True
+    )
+
+    # Find all running worker containers who's Project and Job was deleted from the database
+    worker_without_job_ids = set(worker_ids) - set(worker_with_job_ids)
+
+    for worker_id in worker_without_job_ids:
+        container = client.containers.get(worker_id)
+        container.kill()
+        container.remove()
```

Comment:
Maybe we could add another service that takes additional care of cleaning the worker docker layer and does periodically some kind of:

```
def prune_workers():
    client: DockerClient = docker.from_env()
    # Delete stopped worker containers
    # client.containers.prune(filters={"label": "worker"})
    client.containers.prune()
    # TODO not working?
    #         filters={
    #        "label": "worker"
    #    }
```